crates/pixi_git/src/url.rs

```diff
@@ -31,7 +31,12 @@ impl CanonicalUrl {
 
         // Strip credentials.
         let _ = url.set_password(None);
-        let _ = url.set_username("");
+
+        // Only stripping the username if the scheme is not `ssh`. This is because `ssh` URLs should
+        // have the `git` username.
+        if !url.scheme().contains("ssh") {
```

Comment:
@nichmor this was where the issue was coming from as this was stripping `git@` from all URLs given to the fetch.